### ISSUE TITLE
[macOS][Intel] SSHFS likes to crash during AOSP/Halium repo sync

### ISSUE THREAD

**Issue by fredldotme** [canonical/multipass#2270]
**Describe the bug**
Syncing a whole AOSP/Halium tree to a (case-sensitive) external drive on the macOS host causes sshfs to die, leaving the VM with a broken mounted host directory.

**To Reproduce**
1. Set up case-sensitive storage (preferably external) on the host for pulling the AOSP/Halium source code
2. Mount the external volume into a Multipass instance based on Ubuntu 18.04
3. Launch the instance and follow the *Build instructions* here: https://github.com/Halium/projectmanagement/issues/98
4. Watch `repo sync` cause the sshfs process to crash, leaving the VM with a broken mount

**Expected behavior**
Be able to repo sync and build Halium within the VM

**Logs**
Will follow as soon as I get around follow-up issues with Multipass in that scenario (probably issues with regards to cache invalidation between the guest and the host after `rm -rf .repo` inside the VM). But this is driving me nuts so I went ahead and opened the issue anyway.

**Additional info**
 - OS: macOS 11.x and 12.0 beta 8
- `multipass  1.7.2+mac`
- `multipassd 1.7.2+mac`
```
Name:           build-bionic
State:          Running
IPv4:           192.168.64.2
Release:        Ubuntu 18.04.6 LTS
Image hash:     5c410839a464 (Ubuntu 18.04 LTS)
Load:           1.09 0.80 0.36
Disk usage:     2.2G out of 4.7G
Memory usage:   216.1M out of 15.7G
Mounts:         /Volumes/Macintosh External HD 2/Projects => /home/ubuntu/Projects
                    UID map: 501:1000
                    GID map: 20:1000
 ```

**Additional context**
Only tested with external storage as a mount target, as my internal storage is too small to hold a full Halium copy.

**Comment by ricab** [comment]
Hi @fredldotme, just a hunch, but could you be hitting a `maxfiles` limit in macOS? We [saw that before](https://github.com/canonical/multipass/issues/1933#issuecomment-765736094) and it was a mere 256 by default (not sure if that's changed). What does `launchctl limit maxfiles` say? Here is [one guide](https://gist.github.com/devinrhode2/4cbf7f02a9701510d61f5be0515b8286) for changing that. There should be some indication in the logs if this is indeed the problem.

**Comment by underrun** [comment]
I'm having similar issues that only started after upgrading mac to `macOS 11.6 (20G165)` - 

@fredldotme when you say you're "relying on the hosts sshd right now" - what exactly do you mean? you aren't using multipass built in mount functionality and manually setting up sshfs to host?

**Comment by fredldotme** [comment]
@underrun yes, I enabled remote login in macOS's system settings and use `user@192.168.64.1:/path/to/directory` as the mount target in an fstab entry now.
Still has its own quirks which cause me to do the repo sync from outside the VM but it's better than nothing.

**Comment by levkropp** [comment]
Closing due to a workaround existing, as well as MacOS versions 11 and 12 no longer being supported.
